### ISSUE TITLE
Auto increment the ID value when the value is null

### ISSUE THREAD

**Issue by cwittenb** [vuex-orm/vuex-orm#168]
I have encountered an unexpected behaviour trying to insert nested data into the store - all objects where just created, so they do not have an ID yet.

Here are my models:

```
export default class Board extends Model {
  static entity = 'boards'

  static fields () {
    return {
		id: this.increment(),
		name: this.string('my board'),
		holds: this.hasMany(Hold, 'board_id')
    }
  }
}
  
export default class Hold extends Model {
  static entity = 'holds'

  static fields () {
    return {
      id: this.increment(),
      ip: this.string(),
      board_id: this.number()
    }
  }
```

When I call:
```
Board.dispatch('insertOrUpdate', { data: state.current_board }).then((entities) => { .... })
```

where data equals:
```
{
	"$id": 1,
	"id": 1,
	"name": "bigb",
	"holds": [
		{
			"id": null,
			"ip": "192.168.1.13"
		},
		{
			"id": null,
			"ip": "192.168.1.1",
		},
		{
			"id": null,
			"ip": "192.168.1.2",
		},
		{
			"id": null,
			"ip": "192.168.1.3",
		},
		{
			"id": null,
			"ip": "192.168.1.4",
		},
		{
			"id": null,
			"ip": "192.168.1.5",
		},
		{
			"id": null,
			"ip": "192.168.1.6",
		},
		{
			"id": null,
			"ip": "192.168.1.20",
		},
		{
			"id": null,
			"ip": "192.168.1.10",
		}
	]
}
```

The returned entities looks like this:
```
{
	"boards": [
		{
			"$id": 1,
			"id": 1,
			"name": "bigb",
			"holds": []
		}
	],
	"holds": [
		{
			"$id": 2,
			"id": 2,
			"ip": "192.168.1.10",
			"board_id": 1,
		}
	]
}
```

Query for ID with holds:
```
Board.getters('query')().with('holds').find(entities.boards[0].id) ==>
 {
	"$id": 1,
	"id": 1,
	"name": "bigb",
	"holds": [
		{
			"$id": 2,
			"id": 2,
			"ip": "192.168.1.10",
			"board_id": 1,
		}
	]
}

```

I can not make any sense of this...
Why is only the last Hold created?
Am I doing something wrong?

-- Thank you for this beautiful piece of software!

**Comment by kiaking** [comment]
Hm... Could you try removing `id` field from `holds` to see if that make any difference? It should work with `null` but maybe this is a defect in somewhere and I would like to see if it changes anything!

**Comment by cwittenb** [comment]
Thanks for your quick reply.
Yep, that's it!
I'm passing `Hold()` objects to `insertOrUpdate`. If remove those id fields beforehand (`delete hold.id`) to remove those `id`-fields - it just works as expected.

**Comment by cwittenb** [comment]
... and if I set those ids to `id = undefined`, that also works fine.

**Comment by kiaking** [comment]
Thank you for the confirmation! Now I understand that this is the at least expected behavior right now that Vuex ORM will treat `null` as a valid value. So the id key would not getting incremented.

I guess there's no point of arrowing null as a value when defining `increment` field, so let me fix this to be same behavior as `undefined`!

**Comment by cwittenb** [comment]
Thank you for looking into it! Would be great to have a fix, as the workaround - deleting fields beforehand - feels rather clumsy. :)

**Comment by cwittenb** [comment]
I further noticed that although the number of created objects is correct, the reference ids, ie. board_id is 0 or all objects.
Is this a related problem, or should I create a new one?

**Comment by kiaking** [comment]
Hm... Yeah, that is weird. I'll look into that problem along with this issue too! If it's not related, I'll separate issue for that.

**Comment by kiaking** [comment]
Now this issue is fixed at Vuex ORM 0.26.1! 🎉 
https://github.com/vuex-orm/vuex-orm/releases/tag/v0.26.1

Thanks to @kkyouhei for the awesome fix!
